### ISSUE TITLE
Reconfiguring existing server's kube-apiserver-arg for service account token expiration

### ISSUE THREAD

**Issue by TheOnlyWei** [k3s-io/k3s#10573]
This is more of a question and issue combination, since the configurations I am trying isn't working. I am trying to reduce the `serviceAccountToken`'s JWT token expiration time on an k3s cluster. I have it set to `1000` seconds in my manifest YAML:
```
...
- serviceAccountToken:
    expirationSeconds: 1000
    path: token
...
```

Here is the reference of those parameters for k8s:
```
--service-account-extend-token-expiration Default: true
--service-account-max-token-expiration duration
```
https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

**Environmental Info:**
K3s Version: 
```
aksedge-user@aksee2-ledge [ ~ ]$ k3s -v
k3s version v1.27.6+k3s- ()
go version go1.20.10
```
Node(s) CPU architecture, OS, and Version: 
```
Linux aksee2-ledge 5.15.145.2-1.cm2 #1 SMP Wed Jan 17 15:39:07 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

Cluster Configuration:
1 server, 1 agent

**Describe the bug:**
Trying to reconfigure kube-apiserver-arg arguments of an existing API server to reduce the `serviceAccountToken`'s JWT token expiration time on an k3s cluster.

I have followed this instruction that references updating kube-apiserver-arg arguments of an existing cluster but for `audit-log-path` and `audit-policy-file` configurations:
https://docs.k3s.io/security/hardening-guide#api-server-audit-configuration

and set the following to `/etc/systemd/system/k3s.service` similar to the document instructions above, however it didn't reduce the generated JWT token expiration time, which remains at "1721936003", which translates to about 10 minutes in https://jwt.ms/.
```
ExecStart=/usr/local/bin/k3s \
    server \
        '--kube-apiserver-arg=service-account-extend-token-expiration=false' \
        '--kube-apiserver-arg=service-account-max-token-expiration=1h00m0s' \
        $K3S_SERVER_ARGS \
```

I found this file `/etc/rancher/k3s/k3s.env`, which references another file called `/var/.eflow/config/k3s/k3s-config.yml`. I added to this file the properties `service-account-extend-token-expiration=false` and `service-account-max-token-expiration=1h00m0s` in an attempt to reduce the JWT expiration time, but it didn't work either.

```
...

kube-apiserver-arg:
  ...
  - service-account-extend-token-expiration=false
  - service-account-max-token-expiration=1h00m0s
...
```

**Steps To Reproduce:**
- I set up the k3s server using this document: https://learn.microsoft.com/en-us/azure/aks/hybrid/aks-edge-quickstart

**Expected behavior:**
I couldn't find any documentation on how to generally update kube-apiserver-arg after a cluster has been deployed, so maybe the steps I tried was not expected to work and only works for `audit-log-path` and `audit-policy-file`. But the only document I found that mentioned updating kube-apiserver-arg at all for a deployed cluster didn't work.

**Actual behavior:**
The following should update the JWT token expiration? Again, I am not sure if this is expected to work or if I have to do something else, but I can't find any information on this issue.
```
ExecStart=/usr/local/bin/k3s \
    server \
        '--kube-apiserver-arg=service-account-extend-token-expiration=false' \
        '--kube-apiserver-arg=service-account-max-token-expiration=1h00m0s' \
        $K3S_SERVER_ARGS \
```

**Additional context / logs:**
N/A

**Comment by brandond** [comment]
What you're doing looks fine to me. Are you trying to change it on an existing cluster with running pods? I suspect that won't work.

```console
brandond@dev01:~$ kubectl get node -o yaml | grep node-args
      k3s.io/node-args: '["server","--kube-apiserver-arg","service-account-extend-token-expiration=false","--kube-apiserver-arg","service-account-extend-token-expiration=false"]'

brandond@dev01:~$ kubectl get pod -A -o wide
NAMESPACE     NAME                                     READY   STATUS      RESTARTS   AGE   IP          NODE           NOMINATED NODE   READINESS GATES
kube-system   coredns-77ccd57875-ml96z                 1/1     Running     0          14m   10.42.0.5   k3s-server-1   <none>           <none>
kube-system   local-path-provisioner-957fdf8bc-dfb4p   1/1     Running     0          14m   10.42.0.4   k3s-server-1   <none>           <none>
kube-system   helm-install-traefik-crd-cfjmw           0/1     Completed   0          14m   10.42.0.6   k3s-server-1   <none>           <none>
kube-system   svclb-traefik-e0d51e26-69xtd             2/2     Running     0          13m   10.42.0.7   k3s-server-1   <none>           <none>
kube-system   helm-install-traefik-z6hmw               0/1     Completed   1          14m   10.42.0.2   k3s-server-1   <none>           <none>
kube-system   traefik-64f55bb67d-j7npb                 1/1     Running     0          13m   10.42.0.8   k3s-server-1   <none>           <none>
kube-system   metrics-server-648b5df564-2zh4t          1/1     Running     0          14m   10.42.0.3   k3s-server-1   <none>           <none>

brandond@dev01:~$ kubectl exec -it -n kube-system   local-path-provisioner-957fdf8bc-dfb4p -- cat /run/secrets/kubernetes.io/serviceaccount/token
eyJhbGciOiJSUzI1NiIsImtpZCI6Ii04U19OSVRFbmRma2dOV3BPNlh5ZHh6QkUtZEZZYzg1UkI2STJ3Mm9wdVkifQ.eyJhdWQiOlsiaHR0cHM6Ly9rdWJlcm5ldGVzLmRlZmF1bHQuc3ZjLmNsdXN0ZXIubG9jYWwiLCJrM3MiXSwiZXhwIjoxNzIxOTQxNzk0LCJpYXQiOjE3MjE5MzgxODcsImlzcyI6Imh0dHBzOi8va3ViZXJuZXRlcy5kZWZhdWx0LnN2Yy5jbHVzdGVyLmxvY2FsIiwia3ViZXJuZXRlcy5pbyI6eyJuYW1lc3BhY2UiOiJrdWJlLXN5c3RlbSIsInBvZCI6eyJuYW1lIjoibG9jYWwtcGF0aC1wcm92aXNpb25lci05NTdmZGY4YmMtZGZiNHAiLCJ1aWQiOiI4ZmRhOGYwNS1iYTgzLTQ2MzUtYWEwNC1lYjI4MDQzZGExYjkifSwic2VydmljZWFjY291bnQiOnsibmFtZSI6ImxvY2FsLXBhdGgtcHJvdmlzaW9uZXItc2VydmljZS1hY2NvdW50IiwidWlkIjoiMDNkMmI4ZGUtNGZmOS00MTA1LThlYWItMTI3Mzg2N2UzYmZmIn19LCJuYmYiOjE3MjE5MzgxODcsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDprdWJlLXN5c3RlbTpsb2NhbC1wYXRoLXByb3Zpc2lvbmVyLXNlcnZpY2UtYWNjb3VudCJ9.r8V6DBOKk4tij-ccJvKX8SEFmzpE8D-keQut7AyoKjZ9mNl-Lgtdu-A7iJHEviS0WJqs8nra0KzC-lIYoezIJCqZOEJefjSq_a1J8n6OMywET84i2OpkDGLaJxHibCucV3zc_IAeWA9ftON8RXV7htsPN9ea4Aov6SJKFE3Esv1dhw5PprvoyOAVenPccDm8Lktu3ECQjIySu91ur_wBu-8wbR-gU6OvOeBNQL_JZCYH6TrVeY4WY8xp8cOoQ7f1-iykfAmFvbl1eRIWLMIdrlbuxJpkFSsxnHGlkwejgLRM7DFRWVpWvkmcbhThuedP2UhqBoTCGXHOgypOrwcp-Q
```

That decodes as follows:
![image](https://github.com/user-attachments/assets/d5901ea4-bc48-434c-bbaa-b38c4ec26f03)

Note that the issue and expiration times are 3607 seconds apart. I suspect that there is some internal padding of the window to allow for rotation when they expire.

**Comment by TheOnlyWei** [comment]
@brandond Yes, I am trying to change it on an existing K3s cluster. So, the user needs to add `--service-account-extend-token-expiration` and `--service-account-max-token-expiration` to `--kube-apiserver-arg` during cluster creation? So it should work during cluster creation?

**Comment by brandond** [comment]
The flag appears to be working properly to me, however you should note that its value is clamped to a lower and upper bound of 1 hour and 2^32 seconds, so you're not going to be able to set it any lower than that.

https://github.com/kubernetes/kubernetes/blob/v1.27.6/cmd/kube-apiserver/app/server.go#L579-L585

**Comment by brandond** [comment]
tl;dr
1. You can't set it lower than 1 hour
2. If you want to change it after the fact, you need to delete and recreate any pods that have tokens issued with the old value. Any tokens stored outside the token injected into the pod (via direct use of the TokenRequest API) are still valid with their original duration.
